### ISSUE TITLE
MSan: crash in groupArray

### ISSUE THREAD

**Issue by KochetovNicolai** [ClickHouse/ClickHouse#16020]
```
SELECT groupArrayMerge(1048577)(y * 1048576) FROM (SELECT groupArrayState(9223372036854775807)(x) AS y FROM (SELECT 1048576 AS x))
```
```
2020.10.15 16:52:17.088012 [ 1493 ] {} <Fatal> BaseDaemon: ########################################
2020.10.15 16:52:17.091205 [ 1493 ] {} <Fatal> BaseDaemon: (version 20.10.1.1, build id: 1730D04609489F71) (from thread 1360) (query_id: e15c48d3-1d74-4352-9274-4d6bc86afc6e) Received signal Segmentation fault (11)
2020.10.15 16:52:17.091661 [ 1493 ] {} <Fatal> BaseDaemon: Address: 0x7f392e9d3000 Access: read. Address not mapped to object.
2020.10.15 16:52:17.092151 [ 1493 ] {} <Fatal> BaseDaemon: Stack trace: 0x7f3a402cd6ae 0x10205d20 0x139c65bf 0x139c5f54 0x13a363a6 0x1f30ea38 0x1f30b69e 0x1f307202 0x180cfa0e 0x181e6a9b 0x181eab28 0x3532f097 0x3533747c 0x39cb7d27 0x37740c0e 0x391dbe05 0x39386f4a 0x39386b53 0x393869a1 0x393868c1 0x39386841 0x39382da9 0x10362e8b 0x10362c02 0x3937dc1f 0x3937f9cf 0x3937963e
2020.10.15 16:52:17.095771 [ 1493 ] {} <Fatal> BaseDaemon: 5. /build/glibc-2ORdQG/glibc-2.27/string/../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:370: __memmove_sse2_unaligned_erms @ 0xbb6ae in /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.27.so
2020.10.15 16:52:17.459184 [ 1493 ] {} <Fatal> BaseDaemon: 6. __msan_memcpy @ 0x10205d20 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.496460 [ 1493 ] {} <Fatal> BaseDaemon: 7. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Common/PODArray.h:497: void DB::PODArray<unsigned int, 32ul, DB::MixedArenaAllocator<4096ul, Allocator<false, false>, DB::AlignedArenaAllocator<4ul>, 4ul>, 0ul, 0ul>::insert_assume_reserved<unsigned int const*, unsigned int const*>(unsigned int const*, unsigned int const*) @ 0x139c65bf in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.521386 [ 1493 ] {} <Fatal> BaseDaemon: 8. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Common/PODArray.h:458: void DB::PODArray<unsigned int, 32ul, DB::MixedArenaAllocator<4096ul, Allocator<false, false>, DB::AlignedArenaAllocator<4ul>, 4ul>, 0ul, 0ul>::insert<unsigned int const*, unsigned int const*, DB::Arena*&>(unsigned int const*, unsigned int const*, DB::Arena*&) @ 0x139c5f54 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.555197 [ 1493 ] {} <Fatal> BaseDaemon: 9. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/AggregateFunctions/AggregateFunctionGroupArray.h:235: DB::GroupArrayNumericImpl<unsigned int, DB::GroupArrayTrait<true, (DB::Sampler)0> >::merge(char*, char const*, DB::Arena*) const @ 0x13a363a6 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.629716 [ 1493 ] {} <Fatal> BaseDaemon: 10. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Functions/FunctionBinaryArithmetic.h:671: DB::FunctionBinaryArithmetic<DB::MultiplyImpl, DB::NameMultiply, true>::executeAggregateMultiply(DB::FunctionArguments&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long) const @ 0x1f30ea38 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.698130 [ 1493 ] {} <Fatal> BaseDaemon: 11. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Functions/FunctionBinaryArithmetic.h:1065: DB::FunctionBinaryArithmetic<DB::MultiplyImpl, DB::NameMultiply, true>::executeImpl(DB::FunctionArguments&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long) const @ 0x1f30b69e in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.766227 [ 1493 ] {} <Fatal> BaseDaemon: 12. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Functions/FunctionBinaryArithmetic.h:0: DB::FunctionBinaryArithmeticWithConstants<DB::MultiplyImpl, DB::NameMultiply, true>::executeImpl(DB::FunctionArguments&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long) const @ 0x1f307202 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.809535 [ 1493 ] {} <Fatal> BaseDaemon: 13. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Functions/IFunctionAdaptors.h:153: DB::DefaultExecutable::execute(DB::FunctionArguments&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long) @ 0x180cfa0e in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.849550 [ 1493 ] {} <Fatal> BaseDaemon: 14. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Functions/IFunction.cpp:325: DB::ExecutableFunctionAdaptor::executeWithoutLowCardinalityColumns(std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> >&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long, bool) @ 0x181e6a9b in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:17.885766 [ 1493 ] {} <Fatal> BaseDaemon: 15. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Functions/IFunction.cpp:500: DB::ExecutableFunctionAdaptor::execute(std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> >&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long, bool) @ 0x181eab28 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:18.095622 [ 1493 ] {} <Fatal> BaseDaemon: 16. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Interpreters/ExpressionActions.cpp:358: DB::ExpressionAction::execute(DB::Block&, bool) const @ 0x3532f097 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:18.306149 [ 1493 ] {} <Fatal> BaseDaemon: 17. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Interpreters/ExpressionActions.cpp:621: DB::ExpressionActions::execute(DB::Block&, bool) const @ 0x3533747c in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:18.579755 [ 1493 ] {} <Fatal> BaseDaemon: 18. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Processors/Transforms/ExpressionTransform.cpp:25: DB::ExpressionTransform::transform(DB::Chunk&) @ 0x39cb7d27 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:18.835855 [ 1493 ] {} <Fatal> BaseDaemon: 19. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Processors/ISimpleTransform.h:43: DB::ISimpleTransform::transform(DB::Chunk&, DB::Chunk&) @ 0x37740c0e in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:19.099172 [ 1493 ] {} <Fatal> BaseDaemon: 20. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Processors/ISimpleTransform.cpp:89: DB::ISimpleTransform::work() @ 0x391dbe05 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:19.365968 [ 1493 ] {} <Fatal> BaseDaemon: 21. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Processors/Executors/PipelineExecutor.cpp:78: DB::executeJob(DB::IProcessor*) @ 0x39386f4a in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:19.626984 [ 1493 ] {} <Fatal> BaseDaemon: 22. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Processors/Executors/PipelineExecutor.cpp:95: DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0::operator()() const @ 0x39386b53 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:19.889272 [ 1493 ] {} <Fatal> BaseDaemon: 23. /home/nik-kochetov/dev/ClickHouse/build-msan/../contrib/libcxx/include/type_traits:3519: decltype(std::__1::forward<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&>(fp)()) std::__1::__invoke<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&>(DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&) @ 0x393869a1 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:20.155410 [ 1493 ] {} <Fatal> BaseDaemon: 24. /home/nik-kochetov/dev/ClickHouse/build-msan/../contrib/libcxx/include/__functional_base:349: void std::__1::__invoke_void_return_wrapper<void>::__call<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&>(DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&) @ 0x393868c1 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:20.420388 [ 1493 ] {} <Fatal> BaseDaemon: 25. /home/nik-kochetov/dev/ClickHouse/build-msan/../contrib/libcxx/include/functional:1540: std::__1::__function::__alloc_func<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0, std::__1::allocator<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0>, void ()>::operator()() @ 0x39386841 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:20.685067 [ 1493 ] {} <Fatal> BaseDaemon: 26. /home/nik-kochetov/dev/ClickHouse/build-msan/../contrib/libcxx/include/functional:1714: std::__1::__function::__func<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0, std::__1::allocator<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0>, void ()>::operator()() @ 0x39382da9 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:20.700581 [ 1493 ] {} <Fatal> BaseDaemon: 27. /home/nik-kochetov/dev/ClickHouse/build-msan/../contrib/libcxx/include/functional:1867: std::__1::__function::__value_func<void ()>::operator()() const @ 0x10362e8b in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:20.709604 [ 1493 ] {} <Fatal> BaseDaemon: 28. /home/nik-kochetov/dev/ClickHouse/build-msan/../contrib/libcxx/include/functional:2473: std::__1::function<void ()>::operator()() const @ 0x10362c02 in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:20.970950 [ 1493 ] {} <Fatal> BaseDaemon: 29. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Processors/Executors/PipelineExecutor.cpp:561: DB::PipelineExecutor::executeStepImpl(unsigned long, unsigned long, std::__1::atomic<bool>*) @ 0x3937dc1f in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:21.232758 [ 1493 ] {} <Fatal> BaseDaemon: 30. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Processors/Executors/PipelineExecutor.cpp:477: DB::PipelineExecutor::executeSingleThread(unsigned long, unsigned long) @ 0x3937f9cf in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse
2020.10.15 16:52:21.495786 [ 1493 ] {} <Fatal> BaseDaemon: 31. /home/nik-kochetov/dev/ClickHouse/build-msan/../src/Processors/Executors/PipelineExecutor.cpp:752: DB::PipelineExecutor::executeImpl(unsigned long) @ 0x3937963e in /home/nik-kochetov/dev/ClickHouse/build-msan/programs/clickhouse

```
https://clickhouse-test-reports.s3.yandex.net/15999/40f62719ff5fedc7ee1a41b87e5abef2751638cb/fuzzer/report.html#fail1

It is reproducible with msan, 20.10.1

**Comment by hexiaoting** [comment]
@KochetovNicolai 
This query runs ok with version 20.11.1.1.

**Comment by alexey-milovidov** [comment]
@hexiaoting Did you test with MSan?

**Comment by hexiaoting** [comment]
> @hexiaoting Did you test with MSan?

sorry, I haven't test this query with MSan yesterday.
I try it with MSan today, it failed.

**Comment by hexiaoting** [comment]
@KochetovNicolai 
what does y * 1048576 mean in this query?
multipy a  AggregateFunction(groupArray(262145), UInt8) and UInt32 ???

```
SELECT groupArrayMerge(y * 1048576) FROM (SELECT groupArrayState(262145)(x) AS y FROM (SELECT 1 AS x)) 【failed】
SELECT groupArrayMerge(y * 1048576) FROM (SELECT groupArrayState(262144)(x) AS y FROM (SELECT 1 AS x)) 【sucess】
```
And what's the relationship between 1048576 and 262145???

**Comment by KochetovNicolai** [comment]
@hexiaoting this query was found by fuzz test. There is no meaning in query.
Multiplication of  `AggregateFunction(groupArray(262145), UInt8)` and `UInt32` looks strange indeed. However, no query should crash server.

**Comment by alexey-milovidov** [comment]
@KochetovNicolai Not strange at all. We support adding and multiplying aggregation states in a natural way.

**Comment by alexey-milovidov** [comment]
@hexiaoting Query Fuzzer is a tool that actively searches for random weird corner cases, most of them don't even have any sense :)

**Comment by l1t1** [comment]
> SELECT groupArrayMerge(y * 1048576) FROM (SELECT groupArrayState(262145)(x) AS y FROM (SELECT 1 AS x))

the 1st sql runs ok and returns
```
[1,1,1,1,1,1,1,1,1,1,1,1⋯│

1 rows in set. Elapsed: 0.043 sec.
```

**Comment by alexey-milovidov** [comment]
Maybe fixed in #18526?
